### ISSUE TITLE
"--quiet" flag does not disable stdout output as documented in code

### ISSUE THREAD

**Issue by nmiyake** [vektra/mockery#309]
The current code for handling the "--quiet" flag does the following:

```
// if "quiet" flag is set, set os.Stdout to /dev/null to suppress all output to Stdout
os.Stdout = os.NewFile(uintptr(syscall.Stdout), os.DevNull)
```

However, this does not do what the comment says -- because the file descriptor being provided is `syscall.Stdout` this overrides the `os.DevNull` name and the resulting file still writes to standard output.

This can be seen with `--version` output. The code prints to stdout:

```
if r.Config.Version {
	fmt.Println(config.SemVer)
	return nil
}
```

And `--quiet` does not suppress its output:

```
➜ mockery --version
04 Jul 20 14:33 PDT INF Starting mockery dry-run=false version=0.0.0-dev
0.0.0-dev
➜ mockery --version --quiet
0.0.0-dev
```

As currently written, `--quiet` just turns off the logger output (which is set to go to `stderr`). This may be the intended behavior, but it doesn't match the code or the help documentation.

**Comment by nmiyake** [comment]
If you do want to redirect `os.Stdout` to `/dev/null`, correct code would be along the lines of:

```go
os.Stdout, err = os.Open(os.DevNull)
if err != nil {
	fmt.Fprintf(os.Stderr, "Failed to open %s: %v\n", os.DevNull, err)
	return err
}
```

With this change, writing to `stdout` produces nothing (as expected). Happy to open a PR for this, but would like to clarify the intended behavior of `--quiet` first.

**Comment by LandonTClipp** [comment]
A few thoughts on this:
- The `quiet` flag really shouldn't exist at all. This is left over from v1. Dummy me didn't remove this for v2 so now we're stuck with it :'( I have this for v3 that we will remove it.
- I think what we really should be doing is make `--quiet` just an alias for `--log-level fatal`. Or possibly just disable all logging.

**Comment by nmiyake** [comment]
Got it. Yes, currently it looks like pretty much all of the output is printed to `stderr` anyway (besides when a user is explicitly asking to have output on stdout via commands like `version` or flags like `--print`), so the purported purpose of `--quiet` doesn't make much sense in this context.

Given that, I think the correct fix for the current state is to update the documentation to specify that `--quiet` is equivalent to `--log-level=""` since that's the behavior that actually shipped with v2, and, as you stated, you can remove the flag entirely in v3.

**Comment by LandonTClipp** [comment]
Closed per #314

**Comment by iwittkau** [comment]
Did something change in the meantime?

I still see debug outputs when using `--quiet` or `--log-level fatal`.

 `--log-level=""` results in

```
Failed to initialize logger: Couldn't parse log level: Unknown Level String: 'NoLevel', defaulting to NoLevel
```
